### PR TITLE
Clarify TimeRange validity periods

### PR DESCRIPTION
**Pull request by znewman01**

### CODE REVIEW

**Review by loosebazooka — APPROVED** [review]
nice, thanks!

**Review by haydentherapper — APPROVED** [review]

**Review by kommendorkapten — CHANGES_REQUESTED** [review]
This comments are now in conflict with the definition of a `TimeRange`:

https://github.com/sigstore/protobuf-specs/blob/main/protos/sigstore_common.proto#L190

> // The time range is half-open and does not include the end timestamp,
> // i.e [start, end).

We should pick one definition and be consistent IMO.

I would be in favour of using the already defined (Half open), but are open for discussion.

edit: fixed a broken sentence.

**Comment by kommendorkapten** [comment]
sigstore-js is *inclusive* for both certificate and transparency log verification:
https://github.com/sigstore/sigstore-js/blob/b4e96a2df70abbcb0cc210f50568de04d46c23e8/src/ca/verify/chain.ts#L74
https://github.com/sigstore/sigstore-js/blob/b4e96a2df70abbcb0cc210f50568de04d46c23e8/src/tlog/verify/set.ts#L114

So this may indicate that we should align with RFC 5280.

@woodruffw any details on how Python client?

**Comment by znewman01** [comment]
Good catch. My vote is to align with RFC 5280. I've updated the `TimeRange` comment.

Fortunately, it doesn't *actually* matter—a nanosecond should never make the difference in normal use.

**Review by haydentherapper — APPROVED** [review]

File: protos/sigstore_trustroot.proto [hunk 1]
```diff
@@ -52,7 +52,10 @@ message CertificateAuthority {
         dev.sigstore.common.v1.X509CertificateChain cert_chain = 3;
         // The time the *entire* chain was valid. This is at max the
         // longest interval when *all* certificates in the chain were valid,
-        // but it MAY be shorter.
+	// but it MAY be shorter. Clients MUST check timestamps against *both*
+	// the `valid_for` time range *and* the entire certificate chain.
```
Comment (woodruffw) [inline protos/sigstore_trustroot.proto:56]: Weird spacing/indentation here? 🙂

**Comment by woodruffw** [comment]
> @woodruffw any details on how Python client?

Looks like we actually have a small bug with this: we're inclusive on `notAfter` but not on `notBefore`. I'll fix that now.

**Comment by woodruffw** [comment]
> Good catch. My vote is to align with RFC 5280. I've updated the `TimeRange` comment.

Strong agree!

File: protos/sigstore_trustroot.proto (hunk 1 above)
Comment (znewman01) [reply protos/sigstore_trustroot.proto]: whoops, was tabs

**Review by haydentherapper — APPROVED** [review]

**Review by kommendorkapten — APPROVED** [review]
Nice!

**Pull request merged**